source/extensions/filters/network/common/factory_base.h

```diff
@@ -75,6 +75,57 @@ class FactoryBase : public Server::Configuration::NamedNetworkFilterConfigFactor
   const bool is_terminal_filter_;
 };
 
+/**
+ * Common base class for upstream network filter factory registrations. Removes a substantial
+ * amount of boilerplate.
+ */
+template <class ConfigProto, class ProtocolOptionsProto = ConfigProto>
+class UpstreamFactoryBase : public Server::Configuration::NamedUpstreamNetworkFilterConfigFactory {
```

Comment:
I think this is the first upstream network filter; adding the boiler-plate-reducer, similar to the one for downstream network filters.